pkg/config/system_probe.go

```diff
@@ -131,6 +133,9 @@ func InitSystemProbeConfig(cfg Config) {
 	cfg.BindEnvAndSetDefault(join(spNS, "zypper_repos_dir"), suffixHostEtc(defaultZypperReposDirSuffix), "DD_ZYPPER_REPOS_DIR")
 	cfg.BindEnvAndSetDefault(join(spNS, "attach_kprobes_with_kprobe_events_abi"), false, "DD_ATTACH_KPROBES_WITH_KPROBE_EVENTS_ABI")
 
+	// User Tracer
+	cfg.BindEnvAndSetDefault(join(diNS, "enabled"), false, "DD_DYNAMIC_INSTRUMENTATION_ENABLED")
```

Comment:
The metadata/inventories package seems to have specific features rather than having the whole module enabled. For example USM just has different network protocols it supports, rather than USM as a whole. Are you sure it makes sense to have something like `feature_dynamic_instrumentation_enabled`? I don't know what inventories is so I really don't know.

Similarly, dynamic instrumentation might not belong under `pkg/network/config` since it's not a network feature.

Reply:
usm is not the best example (unforutnately, as we started as a feature in npm and evolved)
but take a look in npm https://github.com/DataDog/datadog-agent/blob/main/pkg/metadata/inventories/inventories.go#L418
or CWS https://github.com/DataDog/datadog-agent/blob/main/pkg/metadata/inventories/inventories.go#L415

Reply:
regarding the configuration test, you don't need to add them in `pkg/network/config/config_test.go` but a similar to the correct `config_test.go` file
another example in other package https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_test.go

Reply:
Sounds good - I've added to the metadata/inventories package and added a test to `cmd/system-probe/config/config_linux_test.go`